bitcoin/src/blockdata/script/mod.rs

```diff
@@ -142,18 +145,60 @@ pub fn write_scriptint(out: &mut [u8; 8], n: i64) -> usize {
     len
 }
 
+/// Returns minimally encoded scriptint as a byte vector.
+pub fn scriptint_vec(n: i64) -> Vec<u8> {
+    let mut buf = [0u8; 8];
+    let len = write_scriptint(&mut buf, n);
+    buf[0..len].to_vec()
+}
+
 /// Decodes an integer in script format without non-minimal error.
 ///
 /// The overflow error for slices over 4 bytes long is still there.
 ///
 /// See [`push_bytes::PushBytes::read_scriptint`] for a description of some subtleties of
 /// this function.
 pub fn read_scriptint_non_minimal(v: &[u8]) -> Result<i64, Error> {
+    read_scriptint_size(v, DEFAULT_MAX_SCRIPTINT_SIZE, false)
+}
+
+/// Decodes an interger in script format with flexible size limit.
+///
+/// Note that in the majority of cases, you will want to use either
+/// [`push_bytes::PushBytes::read_scriptint`] or [`read_scriptint_non_minimal`] instead.
+///
+/// Panics if max_size exceeds 8.
+pub fn read_scriptint_size(v: &[u8], max_size: usize, minimal: bool) -> Result<i64, Error> {
```

Comment:
In fd78f83e7e4f2fb6dcbd5ce8ca555b436973f72b:

Do we really want to expose this in the public API? It seems like it's only used with two values -- 4 and 5 -- and these ought to be special-cased. The 5-byte version is encountered in two places: (a) when parsing locktimes and (b) whe parsing the output of something like OP_ADD with overflow.

Reply:
Removed.